lib/util/source-code.js

```diff
@@ -88,68 +88,66 @@ function looksLikeExport(astNode) {
  * @param {ASTNode} ast - The Program node of the AST representing the code. This AST should be created from the text that BOM was stripped.
  * @constructor
  */
-function SourceCode(text, ast) {
-    validate(ast);
+class SourceCode {
```

Comment:
We can't convert this right now, as its part of our exported API. Converting to a class would be a breaking change.